src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs

```diff
@@ -59,6 +60,19 @@ public DataServiceRestClient( [NotNull] Uri serverUri, int maxUriLength = RestCl
 			: base( restClient ?? new RestClient( serverUri, EndpointName, maxUriLength: maxUriLength, serializer: ObjectSerializer.SystemTextJson ) )
 		{ }
 
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DataServiceRestClient"/> class.
+		/// </summary>
+		/// <param name="serverUri">The PiWeb Server uri, including port and instance</param>
+		/// <param name="maxDegreeOfParallelism"> The maximum number of concurrent tasks enabled by a <see cref="T:System.Threading.Tasks.ParallelOptions" /> instance.</param>
+		/// <param name="maxUriLength">The uri length limit</param>
+		/// <param name="restClient">Custom implementation of RestClient</param>
+		public DataServiceRestClient( [NotNull] Uri serverUri, int maxDegreeOfParallelism, int maxUriLength = RestClientBase.DefaultMaxUriLength, RestClientBase restClient = null )

```

Comment:
I would rename maxDegreeOfParallelism to maxRequestsInParallel and I would switch the parameter order so that the we prevent errors that might happen when somebody is not careful. previously the first int parameter was maxUriLength, now it is maxDegreeOfParallelism

Reply:
Parameter order has been adjusted. Paramater is renamed.